### ISSUE TITLE
uploadAttachment buffer data 

### ISSUE THREAD

**Issue by JonathanZWhite** [Yoctol/messaging-apis#223]
I saw that @chentsulin recently merged in a PR that added `buffer` support for `uploadAttachment`. Thanks for doing that 😄 

I'm trying to use a buffer to upload an image attachment straight to Messenger without saving the file on my server. 

The following function calls work for me:
```js
client.uploadAttachment('image', 'http://www.example.com/image.jpg');
client.uploadAttachment('image', fs.createReadStream('image.jpg'));
```

However, if I pass a `buffer` to `uploadAttachment` I get the following error: 
```
Messenger API - 100 OAuthException (#100) Incorrect number of files uploaded. Must upload exactly one file.
```

For example, if I submit multi-part form data to my server with a `file` and pass the `buffer` to `uploadAttachment` I get the above error.
```js
client.uploadAttachment('image', req.file.buffer);
```

I found this [StackOverflow post](https://stackoverflow.com/a/43914175) that suggests that you need to set a `filename`. I tested adding a `filename` on the following line and it seemed to resolve the issue: https://github.com/Yoctol/messaging-apis/blob/87bb438de1524514a319622f4c614f64bd123af1/packages/messaging-api-messenger/src/MessengerClient.js#L1228

**Comment by JonathanZWhite** [comment]
In order to fix this issue, I think the extension type would also have to be known for the buffer.

```
form.append('filedata', attachment, { filename: '[file_name].[ext]' })
```

This would either require either the consuming client to pass in the file name or to detect the extension of the buffer with a library like [FileType](https://github.com/sindresorhus/file-type). The former seems more reasonable.

**Comment by chentsulin** [comment]
@JonathanZWhite Thanks for the clear issue explanation.

I think we should allow passing `filename` as option:

```js
client.uploadAttachment('image', req.file.buffer, { filename: 'buf.jpg' });
```

And also set good default using FileType.

Would you want to send a pull request for this?

**Comment by JonathanZWhite** [comment]
You bet 😄

**Comment by chentsulin** [comment]
Closed by #226
